sycl/doc/extensions/supported/sycl_ext_oneapi_backend_level_zero.md

```diff
@@ -115,6 +115,7 @@ struct {
 ``` C++
 struct {
   ze_command_queue_handle_t NativeHandle;
+  device Device;

```

Comment:
The commit message indicates that prior to this PR the device associated with the constructed queue is unspecified.  However, the Level Zero interop spec *does* specify which device is associated with the queue:

> The queue is attached to the first device in the passed SYCL context.

Was the code implemented to do that before?  Maybe we don't need this PR?  Though, I do agree that passing a specific device is more friendly.

Reply:
My bad. Yes, it seems to be in line with the old behavior and I will need to make a small adjustment to recover this.

That said, this seems to have been a mistake in the design. `ze_command_queue_handle_t` is created with a context and a device, so assuming that the right device is also the same device as the first device in the SYCL context is both restrictive and prone to error. We are seeing this with SYCL/Plugin/interop-level-zero.cpp in the test suite, as it attempts to recreate a SYCL queue from a native handle, which on systems with multiple L0 devices will cause it to pick the wrong device, causing hard-to-debug errors.

The user (and the aforementioned test) could be required to make sure the context only has the right device, but this seems too restrictive and it could potentially cause problems if the native context was actually created with multiple devices.